sqs/provider.go

```diff
@@ -125,6 +131,32 @@ func (s *Provider) Deprovision(ctx context.Context, deprovisionData providerifac
 }
 
 func (s *Provider) Bind(ctx context.Context, bindData provideriface.BindData) (*domain.Binding, error) {
+	bindParameters := BindParameters{}
+	if len(bindData.Details.RawParameters) > 0 {
+		decoder := json.NewDecoder(bytes.NewReader(bindData.Details.RawParameters))
+		decoder.DisallowUnknownFields()
+		if err := decoder.Decode(&bindParameters); err != nil {
+			var errStr error
+			switch castErr := err.(type) {
+				case *json.UnmarshalTypeError:
+					errStr = fmt.Errorf(
+						"Expected %#v field of bind parameters to be of type %v",
+						castErr.Field,
+						castErr.Type.String(),
+					)
+				default:
+					errStr = castErr
+			}
+			// wat
+
+			return nil, brokerapi.NewFailureResponse(
+				errStr,
+				http.StatusBadRequest,
+				"bad-json-format",
+			)
```

Comment:
style: could we break this error wrapping logic out into a seperate func like `func validationError(err) error` or similar, so that we can keep this `if` block smaller and the error return closer to the call site.